app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
We _are_ leaving the database in a legal state at the end of the process.

It's probably fine if this lives in `Fog`; if we're using non-`Fog` providers, there's no guarantee this operation makes sense.

Re: the second loop thing, I think that means we need a way to tell `Site`s to forget a proxy without triggering a decommission, since we're manually doing that later anyway.